packages/core/src/llm/model/model.ts

```diff
@@ -99,7 +100,20 @@ export class MastraLLM extends MastraLLMBase {
       toolChoice,
       maxSteps,
       onStepFinish: async (props: any) => {
-        await onStepFinish?.(props);
+        try {
+          await onStepFinish?.(props);
+        } catch (e: unknown) {
+          const mastraError = new MastraError(
+            {
+              id: 'LLM_TEXT_ON_STEP_FINISH_CALLBACK_EXECUTION_FAILED',
+              domain: ErrorDomain.LLM,
+              category: ErrorCategory.USER,
```

Comment:
we probably want to do something with props as well but out of my head i don't know how it contains
```suggestion
              category: ErrorCategory.USER,
              details: {
                runId,
                threadId,
                resourceId,
              }
```